phantomgames/Home/Categories/CategoryRepository.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  CategoryRepository.swift
+//  phantomgames
+//
+//  Created by Aphiwe Shozi on 2024/05/15.
+//
+
+#warning(" Will be developed further after concurrency workshop ")
+
+import Foundation
+
+typealias CategoryResult = (String, Result<[Game], APIError>)
+
+protocol CategoryRepositoryType: AnyObject {
+    func fetchAPIImage(completion: @escaping (Result<[CategoryResult], APIError>) -> Void)
+    func fetchCategory(endpoint: String, completion: @escaping (Result<[Game], APIError>) -> Void)
+
+}
+class CategoryRepository: CategoryRepositoryType {
+    
+    
+    func fetchAPIImage(completion: @escaping (Result<[CategoryResult], APIError>) -> Void) {
+        
+        let dispatchGroup = DispatchGroup()
+        var results: [CategoryResult] = []
+        
+        dispatchGroup.enter()
```

Comment:
this is very strange. Why do you need this?

Reply:
in the warning I stated that it falls under concurrency so it will be further developed after the workshop